### ISSUE TITLE
Improve filter UX in Data Node/Scenario Selector

### ISSUE THREAD

**Issue by FlorianJacta** [Avaiga/taipy#1403]
### Description

The UX of creating filters could be better in the Data Node/Scenario selector. It would be better to propose something like this:

- when the filter is on a boolean to have a switch: this simplifies the process of creating a boolean filter immensely,
- when the filter is on a date to have date/time picker.

Providing a default "action" for each filter would enhance the filter UX. For example, the tags filter could have as a default action: "contains". Right now, you need a lot of clicks to get the filter in place.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by ovindu-a** [comment]
Hi, can I work on this?

**Comment by jrobinAV** [comment]
Sure! @ovindu-a 

Thank you for your contribution.

**Comment by ovindu-a** [comment]
Hi @jrobinAV, quick question. Could you tell me if there is a way I can test the changes once I make them?

**Comment by jrobinAV** [comment]
Hi @ovindu-a ,

I believe @namnguyen20999 @FredLL-Avaiga @FabienLelaquais @dinhlongviolin1 would be the right persons to help you on the UX side.

**Comment by ovindu-a** [comment]
Hi, @namnguyen20999 , @FredLL-Avaiga , @FabienLelaquais , @dinhlongviolin1  if possible could someone let me know how to test the changes I make in the ui? I think I can close this issue if I figure it out.

**Comment by FredLL-Avaiga** [comment]
we have tests in backend (python: pytest), frontend (typescript: jest) and end-to-end (python: playwright)
You can look at the existing tests related to the files you impact.

**Comment by ovindu-a** [comment]
@quest-bot embark

**Comment by ovindu-a** [comment]
@FredLL-Avaiga is there a way that I can run the frontend so I can see the changes that I make. I'm having an issue figuring out how to do that by following the docs.

**Comment by FredLL-Avaiga** [comment]
Of course, yes
so the process is:

1. in taipy-gui: npm run build:dev
2. in taipy: npm i (one time is enough)
3. in taipy npm run build:dev

Then create a simple python script that shows your new feature or bug fix (something like doc\gui\examples\controls\button_action.py or others)
run the script
you can debug in the browser

**Comment by FredLL-Avaiga** [comment]
@ovindu-a  I don't see any PR linked to this issue.
Are you still working on this ?
Do you plan to propose a PR soon ?

**Comment by Vaibhav91one** [comment]
Is this Issue up for grabs? @jrobinAV

**Comment by jrobinAV** [comment]
Hello @Vaibhav91one, 

Thank you for your help. I believe it is still available for a contribution. @FabienLelaquais @FredLL-Avaiga Can you confirm?
Thank you,

**Comment by FabienLelaquais** [comment]
You're more than welcome @Vaibhav91one indeed! Thanks!
Note that this code is used in both the Scenario and the Data Node selectors...

**Comment by FredLL-Avaiga** [comment]
And in any table

**Comment by FabienLelaquais** [comment]
> And in any table

Absolutely! TableFilter (taipy/frontend/taipy-gui/src/components/Taipy) is the real target here.

**Comment by Vaibhav91one** [comment]
Got it. Will see to it.
